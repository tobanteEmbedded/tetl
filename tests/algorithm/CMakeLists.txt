project(algorithm)

tetl_add_test(${PROJECT_NAME} "all_of")
tetl_add_test(${PROJECT_NAME} "binary_search")
tetl_add_test(${PROJECT_NAME} "copy")
tetl_add_test(${PROJECT_NAME} "count")
tetl_add_test(${PROJECT_NAME} "equal")
tetl_add_test(${PROJECT_NAME} "fill")
tetl_add_test(${PROJECT_NAME} "find")
tetl_add_test(${PROJECT_NAME} "for_each")
tetl_add_test(${PROJECT_NAME} "generate")
tetl_add_test(${PROJECT_NAME} "includes")
tetl_add_test(${PROJECT_NAME} "lower_bound")
tetl_add_test(${PROJECT_NAME} "merge")
tetl_add_test(${PROJECT_NAME} "minmax")
tetl_add_test(${PROJECT_NAME} "misc")
tetl_add_test(${PROJECT_NAME} "mismatch")
tetl_add_test(${PROJECT_NAME} "move")
tetl_add_test(${PROJECT_NAME} "partition")
tetl_add_test(${PROJECT_NAME} "remove")
tetl_add_test(${PROJECT_NAME} "replace")
tetl_add_test(${PROJECT_NAME} "reverse")
tetl_add_test(${PROJECT_NAME} "rotate")
tetl_add_test(${PROJECT_NAME} "search")
tetl_add_test(${PROJECT_NAME} "sort")
tetl_add_test(${PROJECT_NAME} "swap_ranges")
tetl_add_test(${PROJECT_NAME} "transform")
tetl_add_test(${PROJECT_NAME} "unique")
tetl_add_test(${PROJECT_NAME} "upper_bound")
