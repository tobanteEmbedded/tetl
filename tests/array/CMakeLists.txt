project(array)

tetl_add_test(${PROJECT_NAME} array)
