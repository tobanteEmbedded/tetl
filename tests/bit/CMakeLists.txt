project(bit)

tetl_add_test(${PROJECT_NAME} bit_cast)
tetl_add_test(${PROJECT_NAME} bit_functions)
