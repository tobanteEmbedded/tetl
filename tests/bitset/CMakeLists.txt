project(bitset)

tetl_add_test(${PROJECT_NAME} bitset)
