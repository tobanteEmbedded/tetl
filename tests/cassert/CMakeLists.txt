project(cassert)

tetl_add_test(${PROJECT_NAME} assert WILL_FAIL TRUE)

# Test default assert_handler. Don't use `tetl_add_test`
add_executable(test_cassert_assert_handler "assert_handler.t.cpp")
target_compile_definitions(test_cassert_assert_handler PRIVATE TETL_DEBUG=1)
target_link_libraries(test_cassert_assert_handler PRIVATE tetl::etl_headers tetl::compiler_warnings tetl::code_coverage)
add_test(NAME "cassert.hpp: assert_handler" COMMAND test_cassert_assert_handler)
set_tests_properties("cassert.hpp: assert_handler" PROPERTIES WILL_FAIL TRUE)
