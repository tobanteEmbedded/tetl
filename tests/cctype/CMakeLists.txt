project(cctype)

tetl_add_test(${PROJECT_NAME} cctype)
