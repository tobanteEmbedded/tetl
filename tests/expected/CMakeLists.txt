project(expected)

tetl_add_test(${PROJECT_NAME} unexpected)
